### DISCUSSION TITLE
New configuration option to make it possible to adjust the order price

### DISCUSSION THREAD

**Discussion by junyuanz1** [brndnmtthws/thetagang#516] [Ideas]
My understanding is that the thetagang program will use midpoint prices to submit orders.

For a contract with a large spread. I normally use the ask price for sell order and bid price for buy order, I would even add a few basic points to count for sudden stock movement. 

Based on this, I would like to introduce a new change to the configuration file:

```
# Price strategy determines how the system calculates the price for orders. 
# Select the mode that best aligns with your trading strategy, market conditions, 
# and risk tolerance.
#
# Valid options:
#   - midpoint: The system automatically determines the price. It may use the midpoint 
#               or adjust based on the configurations:
#               `symbol.<symbol>.adjust_price_after_delay = true` and 
#               `orders.price_update_delay`.
#   - spread_capture: Uses the ask price for sell orders and the bid price for buy orders. 
#                     This is suitable for scenarios where the bid/ask spread is large 
#                     and you aim to capture sudden underlying movements. This mode 
#                     disregards `symbol.<symbol>.adjust_price_after_delay` and 
#                     `orders.price_update_delay` configurations.
#
# The pricing mode can be overridden for specific symbols using:
# `symbol.<symbol>.price_strategy`.
#
price_strategy = "midpoint"

...

  [symbols.SPY]
  ...
  # The bid/ask spread may be significant for certain contracts. 
  # Setting the pricing mode to "greedy" will submit orders using the ask price for sells 
  # and the bid price for buys, bypassing the midpoint calculation. 
  # This approach is ideal for capturing sudden price movements but may involve higher 
  # transaction costs due to the wide spread.
  price_strategy = "spread_capture"

```

Let me know if you have any questions, and I can implement this feature.

**Comment by brndnmtthws** [comment]
I like this, I think it's a good proposal.

As far as getting orders filled without thetagang, I normally pick the midpoint price, and walk it in from there (depending on how badly I want it to fill). Thetagang tries to emulate the same behaviour. There's a balance between 1) getting the order filled and 2) getting a decent price. If you're too conservative with choosing a limit price, the order won't fill, and if you're too aggressive then you can get a bad price. This usually isn't a big deal for anything with a lot of liquidity, but for anything with poor liquidity it can' have a significant impact.

I think for anything with lower liquidity, you are generally better off going with a the midpoint, and also it helps to do combos or spreads to get better execution (i.e., if someone else submits a market order on one of the legs, you're likely to get a fill).

IBKR's adaptive algo orders also do an o.k. job of handling order execution, but it's not a silver bullet and I assume those orders frequently route to market makers (i.e., payment for order flow) which are taking a small cut of the spread for themselves.

I have thought about making this configurable, but just never got around to it.

**Reply by junyuanz1** [reply]
Sounds good. I will implement this after improving several things first :).
